### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by thwint**
Fix for #12 
- adding additional properties to package.json
- changed default to latest supported node and npm for newer Apikana versions

### CODE REVIEW

**Review by ljucam — APPROVED** [review]

File: apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java [hunk 1]
```diff
@@ -158,10 +158,18 @@ protected boolean isPom() {
     protected void generatePackageJson(String version) throws IOException {
         updateJson(working("package.json"), pack -> {
             pack.put("name", mavenProject.getArtifactId());
+            pack.put("version", mavenProject.getVersion());
```
Comment (hiddenalpha) [inline apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java:161]: I strongly suggest to NOT doing this. See https://github.com/swisspush/apikana-java/pull/8

File: apikana-maven-plugin/src/main/java/org/swisspush/apikana/GenerateMojo.java [hunk 2]
```diff
@@ -51,13 +51,13 @@
...
      */
-    @Parameter(defaultValue = "v7.5.0", property = "apikana.node-version")
+    @Parameter(defaultValue = "v10.23.0", property = "apikana.node-version")
     private String nodeVersion;
 
     /**
      * The npm version to be used.
      */
-    @Parameter(defaultValue = "4.2.0", property = "apikana.npm-version")
+    @Parameter(defaultValue = "6.14.8", property = "apikana.npm-version")
```
Comment (hiddenalpha) [inline apikana-maven-plugin/src/main/java/org/swisspush/apikana/GenerateMojo.java:60]: I suggest to NOT use npm 6.x.x at least as long [that issue](https://github.com/npm/npm/issues/20434) is not fixed.

File: apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java [hunk 3]
```diff
@@ -158,10 +158,18 @@ protected boolean isPom() {
...
             scripts.put("apikana", "apikana");
             final Map<String, String> devDependencies = (Map) pack.merge("devDependencies", new HashMap<>(), (oldVal, newVal) -> oldVal);
+            final Map<String,List<String>> customConfig = (Map) pack.merge("customConfig", new HashMap<>(), (oldVal, newVal) -> oldVal);
+            final List<String> plugins = new ArrayList<>();
+            plugins.add("maven");
+            plugins.add("readme");
+            customConfig.put("plugins", plugins);
+            pack.put("customConfig", customConfig);
             devDependencies.put("apikana", version);
+            devDependencies.put("apikana-defaults", "0.0.0");
```
Comment (hiddenalpha) [inline apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java:172]: Where is the source code for this "*apikana-defaults 0.0.0*" hosted? I'm unable to locate it.

File: apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java (hunk 1 above)
Comment (thwint) [reply apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java]: Projects not using semver can now override the version using apikana.apiversion

File: apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java (hunk 3 above)
Comment (thwint) [reply apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java]: For organizations defining defaults in corporate registry  these can be overridden by defining apikana.defaults.version

**Review by hiddenalpha — APPROVED** [review]

**Pull request merged**